app/controllers/projects/samples_controller.rb

```diff
@@ -2,23 +2,20 @@
 
 module Projects
   # Controller actions for Samples
-  class SamplesController < ApplicationController
+  class SamplesController < Projects::ApplicationController
     before_action :sample, only: %i[show edit update destroy]
     before_action :project
     before_action :context_crumbs
+    before_action :authorize_view_project!, only: %i[show index]
+    before_action :authorize_modify_project!, only: %i[new edit]
+
     layout 'projects'
```

Comment:
Since you are basing this off of Projects::ApplicationController you can remove the layout line.

Reply:
Updated in 6dc4a9d5114f8c5f8ef460d3b5108c373769273e

Reply:
You missed this one in your commit